freeipa/src/main/java/com/sequenceiq/freeipa/repository/StackRepository.java

```diff
@@ -59,6 +59,14 @@
     @Query("SELECT s FROM Stack s WHERE s.accountId = :accountId AND s.environmentCrn = :environmentCrn")
     List<Stack> findMultipleByEnvironmentCrnAndAccountIdEvenIfTerminated(@Param("environmentCrn") String environmentCrn, @Param("accountId") String accountId);
 
+    @Query("SELECT s FROM Stack s "
+            + "LEFT JOIN FETCH s.instanceGroups ig "
+            + "LEFT JOIN FETCH ig.instanceMetaData "
```

Comment:
could you drop these? I'm thinking about removing this from the normal list endpoint too, as I see it's causing long running transaction issues
the same for `findMultipleParentByEnvironmentCrnEvenIfTerminatedWithList`

Reply:
In January, lets discuss this. I think there is a good use case to have the metadata provided and we can discuss the performance.... On another note, I just pushed a new change that actually returns more info. When I investigated this, I noticed that the deleted metadata was not returned. But, lets discuss it in January.

Reply:
sure. I'm noting this as I have observed long running transactions, when multiple freeipas are listed, so we should avoid joins if possible as much as we can. In core we had multiple issues from these joins and I don't want to make the same mistakes in FMS.

Reply:
Based on our discussion, this seems to be less of a concern. This query will perform a join for all of the instances for a single environment. This may include multiple clusters of FreeIPA if FreeIPA has been deleted and recovered. But this should be very similar performance to the normal query of a single cluster of FreeIPA.